src/diffusers/models/attention_processor.py

```diff
@@ -2439,57 +2440,58 @@ def __call__(
         for current_ip_hidden_states, scale, to_k_ip, to_v_ip, mask in zip(
             ip_hidden_states, self.scale, self.to_k_ip, self.to_v_ip, ip_adapter_masks
         ):
-            if mask is not None:
-                if not isinstance(scale, list):
-                    scale = [scale]
+            if scale > 0:
```

Comment:
as before: when using image masking, the user can provide a list of masks for each IP Adapter; each mask will have its own scale

code would break here, as you cannot compare a (possible) list with an integer

Reply:
I didn't test this, this part doesn't have a unit test? If you need help to test this, just tell me and I can help too. 

To be honest I haven't tested any of the `multiple images with multiple masks and scales per image`  but I plan to do it soon to make another guide with them.

Reply:
there is just a slow test (test_ip_adapter_multiple_masks_one_adapter in tests/pipelines/ip_adapters/test_ip_adapter_stable_diffusion.py) for the combination `multiple images + masks + scales for IP Adapter`, unfortunately there are no fast tests

Reply:
is this the test case we are talking about?
```python
import torch
from transformers import CLIPVisionModelWithProjection

from diffusers import DPMSolverMultistepScheduler, StableDiffusionXLPipeline
from diffusers.image_processor import IPAdapterMaskProcessor
from diffusers.utils import load_image, logging
from diffusers.utils.logging import set_verbosity


set_verbosity(logging.ERROR)  # to not show cross_attention_kwargs...by AttnProcessor2_0 warnings


# load & process masks
composition_mask = load_image(
    "https://huggingface.co/datasets/OzzyGT/testing-resources/resolve/main/1024_whole_mask.png"
)
female_mask = load_image(
    "https://huggingface.co/datasets/OzzyGT/testing-resources/resolve/main/ip_adapter_None_20240321125641_mask.png"
)
male_mask = load_image(
    "https://huggingface.co/datasets/OzzyGT/testing-resources/resolve/main/ip_adapter_None_20240321125344_mask.png"
)
background_mask = load_image(
    "https://huggingface.co/datasets/OzzyGT/testing-resources/resolve/main/ip_adapter_6_20240321130722_mask.png"
)
processor = IPAdapterMaskProcessor()
masks1 = processor.preprocess([composition_mask], height=1024, width=1024)
masks2 = processor.preprocess([female_mask, male_mask, background_mask], height=1024, width=1024)
masks2 = masks2.reshape(1, masks2.shape[0], masks2.shape[2], masks2.shape[3])  # output -> (1, 3, 1024, 1024)
masks = [masks1, masks2]

# load images
ip_composition_image = load_image(
    "https://huggingface.co/datasets/OzzyGT/testing-resources/resolve/main/ip_adapter__20240321125152.png"
)
ip_female_style = load_image(
    "https://huggingface.co/datasets/OzzyGT/testing-resources/resolve/main/ip_adapter__20240321125625.png"
)
ip_male_style = load_image(
    "https://huggingface.co/datasets/OzzyGT/testing-resources/resolve/main/ip_adapter__20240321125329.png"
)
ip_background = load_image(
    "https://huggingface.co/datasets/OzzyGT/testing-resources/resolve/main/ip_adapter__20240321130643.png"
)


image_encoder = CLIPVisionModelWithProjection.from_pretrained(
    "h94/IP-Adapter", subfolder="models/image_encoder", torch_dtype=torch.float16
).to("cuda")

pipeline = StableDiffusionXLPipeline.from_pretrained(
    "RunDiffusion/Juggernaut-XL-v9", torch_dtype=torch.float16, image_encoder=image_encoder, variant="fp16"
).to("cuda")

pipeline.scheduler = DPMSolverMultistepScheduler.from_config(pipeline.scheduler.config)
pipeline.scheduler.config.use_karras_sigmas = True

pipeline.load_ip_adapter(
    ["ostris/ip-composition-adapter", "h94/IP-Adapter"],
    subfolder=["", "sdxl_models"],
    weight_name=[
        "ip_plus_composition_sdxl.safetensors",
        "ip-adapter_sdxl_vit-h.safetensors",
    ],
    image_encoder_folder=None,
)
pipeline.set_ip_adapter_scale([1.0, [0.75, 0.75, 0.3]])



prompt = "high quality, cinematic photo, cinemascope, 35mm, film grain, highly detailed"
negative_prompt = "anime, cartoon"


image = pipeline(
    prompt=prompt,
    negative_prompt="",
    ip_adapter_image=[ip_composition_image, [ip_female_style, ip_male_style, ip_background]],
    cross_attention_kwargs={"ip_adapter_masks": masks},
    guidance_scale=6.5,
    num_inference_steps=25,
).images[0]

image.save("yiyi_test_mask_multi_out.png")
```

Reply:
Thanks for your test case, I will test with it and update the test codes when the result is as expected.

Reply:
Won't it be better to have a fast test for this?

Reply:
> is this the test case we are talking about?

exactly

Reply:
> Won't it be better to have a fast test for this?

I can create a new PR to add a fast test and update documentation

Reply:
That should work. Thanks as always.